src/pemfile.rs

```diff
@@ -53,7 +51,57 @@ pub enum Item {
     Csr(CertificateSigningRequestDer<'static>),
 }
 
+impl Item {
+    fn from_kind(kind: pem::SectionKind, data: Vec<u8>) -> Option<Self> {
```

Comment:
This method should be on the bottom, since both `from_slice()` and `from_buf()` depend on it. Probably also sort `from_slice()` after `from_buf()`?